src-tauri/src/node.rs

```diff
@@ -52,11 +53,49 @@ impl NativeExecutionDispatch for ExecutorDispatch {
 }
 
 #[tauri::command]
-pub(crate) async fn start_node(path: String, node_name: String) {
-    init_node(path.into(), node_name).await.unwrap();
+/// starts a new node instance
+/// if there is a node instance running previously,
+/// first it stops the previous instance, then starts a new instance
+pub(crate) async fn start_node(path: String, node_name: String) -> bool {
+    static NODE_HANDLE: Mutex<Option<JoinHandle<()>>> = Mutex::const_new(None);
+
+    let mut node_handle_guard = NODE_HANDLE.lock().await;
+
+    // if there is already a node running, stop it
+    if let Some(guard) = node_handle_guard.take() {
+        guard.abort();
+    }
+
+    // wait for the previous instance to finish if new instance is unable to start
+    // we cannot know the previous node instance is completely finished for now
+    // track the issue from here: https://github.com/paritytech/substrate/issues/11654 and https://github.com/subspace/subspace/issues/578
+    if let Err(_) = timeout(Duration::from_secs(10), async {
```

Comment:
This should be a fallback with TODO to remove in the future. I'd argue it is still helpful to have a loop with upgrade step too, it feels wrong to hit an error until you don't as the default behavior.